Fix for Issue 8701

StormNinja17 opened the pull request:
We added some basic code that checks if a Twenty user exists in the Postgre database. If the user doesn't exist, we create it and try to give it necessary permissions. If we are unable to give it the necessary permissions, we log that issue to the console and redirect users to the Twenty documentation. We updated the documentation with a command users can run to manually grant permissions to the new Twenty user.

Close #8701

prastoin commented on packages/twenty-website/src/content/developers/local-setup.mdx:
````diff
@@ -117,6 +117,10 @@ You should run all commands in the following steps from the root of the project.
     ```
+    Note: In some cases when using Docker, you may need to manually grant permissions to the `twenty` user.
+    ```bash
+      psql postgres -c "GRANT ALL PRIVILEGES ON DATABASE \"default\" TO twenty;" -c "GRANT ALL PRIVILEGES ON DATABASE \"test\" TO twenty;"
````
***Remark:*** Would recommend something agnostically of the sub databases like
```
psql postgres -c CREATE USER postgres WITH SUPERUSER PASSWORD password
```

prastoin commented on packages/twenty-website/src/content/developers/local-setup.mdx:
````diff
@@ -148,6 +156,10 @@ You should run all commands in the following steps from the root of the project.
     ```bash
       make postgres-on-docker
     ```
+    Note: In some cases when using Docker, you may need to manually grant permissions to the `twenty` user.
````
***Remark:*** Duplicated docs section, should be centralized

prastoin reviewed (commented):
Hey @StormNinja17 and all ! Hope you're doing great
Unless I'm mistaken as it is script have several bugs ( also checking ci status )

Overall regarding the `entrypoint.sh` updates, I don't think they can work at all.
In the first place if the user does not exist we won't be able to connect to the database the setup it. As by default the setup authentication is `postgres:postgres` the user we wanna create here.

We should be logging recommendation related to suggested added documentation

StormNinja17 commented:
@prastoin Thank you for the feedback!

I switched all references of a "twenty" user to "postgres". Instead of attempting to grant all privileges, I modified the script to attempt to make a superuser instead. I also consolidated the troubleshooting documentation like you requested.

prastoin reviewed (commented):
Hello @StormNinja17, unfortunately as said above i don't think this can work.
The `postgres` user does not exist occurs because that's the `docker-compose.yml` default database credentials. This means that we could transpile your code as:
```
PGPASSWORD=postgres psql -h whatever -p 5432 -U postgres -d postgres -tc "SELECT 1 FROM pg_roles WHERE rolname='postgres'"
```
Is paradoxical

What we should do is trying to perform a very basic query using the provided credentials, such as a `SELECT 1`.

prastoin commented:
Rebased on lastest `entrypoint.sh`

prastoin commented:
Hey @StormNinja17 closing this from the moment feel free to reopen or reach out on discord if you need further information or wanna change issue for your work group
